packages/gatsby-theme-docs/src/utils/navigate-with-filters.js

```diff
@@ -0,0 +1,33 @@
+import { navigate } from '@reach/router';
+import { encode } from 'qss';
+import extractQueryParameters from './extract-query-parameters';
+
+function navigateWithFilters(filters, location) {
+  const currentQueryParameters = extractQueryParameters(location);
+  const newQueryParameters = removeUnnecessaryParameters({
+    ...currentQueryParameters,
+    ...filters,
+  });
```

Comment:
I don't think there is a need to manually remove those params, the `encode` function does this for you.

<img width="268" alt="image" src="https://user-images.githubusercontent.com/1110551/81946961-47621c80-9600-11ea-9c0f-b0b5bca4e76e.png">